packages/peregrine/lib/talons/SearchPage/useSearchPage.js

```diff
@@ -25,46 +26,51 @@ export const useSearchPage = props => {
     } = props;
 
     const sortProps = useSort();
-
     const [currentSort] = sortProps;
     const { sortAttribute, sortDirection } = currentSort;
-
     // Keep track of the sort criteria so we can tell when they change.
     const previousSort = useRef(currentSort);
 
+    // get the URL query parameters.
+    const location = useLocation();
+    const { search } = location;
+    // Keep track of the search terms so we can tell when they change.
+    const previousSearch = useRef(search);
+
     // Set up pagination.
     const [paginationValues, paginationApi] = usePagination();
     const { currentPage, totalPages } = paginationValues;
     const { setCurrentPage, setTotalPages } = paginationApi;
 
     // retrieve app state and action creators
-    const [appState, appApi] = useAppContext();
-    const { searchOpen } = appState;
-    const { toggleDrawer, toggleSearch } = appApi;
+    const [, appApi] = useAppContext();
+    const { toggleDrawer } = appApi;
 
-    // get the URL query parameters.
-    const location = useLocation();
-    const { search } = location;
     const inputText = getSearchParam('query', location);
 
-    // Keep track of the search terms so we can tell when they change.
-    const previousSearch = useRef(search);
+    const searchCategory = useMemo(() => {
+        const inputFilters = getFiltersFromSearch(search);
+        if (inputFilters.size === 0) {
+            return null;
+        }
+
+        const targetCategoriesSet = inputFilters.get('category_id');
+        if (!targetCategoriesSet) {
+            return null;
+        }
+
+        console.log('the set', targetCategoriesSet);
```

Comment:
Whoops!

```suggestion
```

Reply:
😅  I really need to write myself a lint rule for this.

https://github.com/magento/pwa-studio/pull/2685/commits/ce200bfdd87aa9a9a4ff98dfb631f2a8a64d8298